File: 3 JavaScript/solutions/pick6.js
function generateTicket() {
    low = 1;
    high = 99;
    const ticket = [];
    for (let i = 0; i < 6; i++)
        ticket.push(Math.floor(Math.random() * (1 + high - low)) + low);    // see this explanation: https://teamtreehouse.com/community/mathfloor-mathrandom-max-min-1-min-explanation
    return ticket;
}


function compareTickets(winner, ticket) {
    let matches = 0;
    winner.forEach(function (value, index) {
        if (value == ticket[index])
            matches++;
    })
    return matches;
}


function main() {
    const prizes = {
        0: 0,
        1: 4,
        2: 7,
        3: 100,
        4: 50000,
        5: 1000000,
        6: 25000000,
    }
    

    let winnings = 0;
    let expenses = 0;
    let iter = 0;
    const MAX_ITER = 99999;


    const winner = generateTicket();


    while (iter <= MAX_ITER) {
        expenses += 2;
        winnings += prizes[compareTickets(winner, generateTicket())];
        iter++;
    }


    console.log(`\nwinnings: ${winnings}\nexpenses: ${expenses}\nroi: ${(winnings - expenses) / expenses}\n`)
}

main()

File: 3 JavaScript/solutions/pick6.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define MAX_ITER 100000
#define TICKET_LENGTH 6

void generateTicket(int *ticket);
int compareTickets(int *ticket1, int *ticket2);

int main() {
    int winnings = 0, expenses = 0, nIter = 0;
    float roi;
    int winningTicket[TICKET_LENGTH];
    int ticket[TICKET_LENGTH];

    srand(time(NULL));

    int prizes[] = {0, 4, 7, 100, 50000, 1000000, 25000000};

    generateTicket(winningTicket);

    while (nIter < MAX_ITER) {
        expenses += 2;
        nIter++;

        generateTicket(ticket);
        winnings += prizes[compareTickets(winningTicket, ticket)];
    }

    printf("\nwinnings: %d\nexpenses: %d\nroi: %f\n\n", winnings, expenses, (float)(winnings - expenses) / expenses);

    return 0;
}


void generateTicket(int *ticket) {
    for(int i = 0; i < TICKET_LENGTH; i++) 
        *(ticket + i) = rand() % 99 + 1;
}


int compareTickets(int *ticket1, int *ticket2) {
    int i, matches = 0;
    for (i = 0; i < TICKET_LENGTH; i++) {
        if (ticket1[i] == ticket2[i])
            matches++;
    }
    return matches;
}